File: frontend/mini_quiz_app/lib/models/response.dart
class Answer {
  final String questionId;
  final dynamic answer;
  Answer({required this.questionId, required this.answer});

  Map<String, dynamic> toJson() => {'questionId': questionId, 'answer': answer};
}

class QuizResponse {
  final String id;
  final String? participantName;
  final DateTime submittedAt;
  final List<Map<String, dynamic>> answers;
  QuizResponse({
    required this.id,
    this.participantName,
    required this.submittedAt,
    required this.answers,
  });

  factory QuizResponse.fromJson(Map<String, dynamic> j) {
    return QuizResponse(
      id: j['_id'] ?? '',
      participantName: j['participantName'],
      submittedAt: DateTime.tryParse(j['submittedAt']?.toString() ?? '') ?? DateTime.now(),
      answers: (j['answers'] as List<dynamic>?)?.map((e) => Map<String, dynamic>.from(e)).toList() ?? [],
    );
  }
}

File: backend/models/Response.js
const mongoose = require('mongoose');

const AnswerSchema = new mongoose.Schema({
  questionId: { type: mongoose.Schema.Types.ObjectId, required: true },
  answer: { type: mongoose.Schema.Types.Mixed } // string or index/option
}, { _id: false });

const ResponseSchema = new mongoose.Schema({
  quizId: { type: mongoose.Schema.Types.ObjectId, ref: 'Quiz', required: true },
  participantName: { type: String }, // optional
  submittedAt: { type: Date, default: Date.now },
  answers: { type: [AnswerSchema], default: [] }
}, { timestamps: false });

module.exports = mongoose.model('Response', ResponseSchema);
